myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java

```diff
@@ -189,23 +175,51 @@ public TaskInfo createTask(Offer offer, FrameworkID frameworkId, TaskID taskId,
       CommandInfo commandInfo = getCommandInfo(serviceProfile, ports);
       ExecutorInfo executorInfo = getExecutorInfoForSlave(frameworkId, offer, commandInfo);
 
-      TaskInfo.Builder taskBuilder = TaskInfo.newBuilder().setName("task-" + taskId.getValue()).setTaskId(taskId).setSlaveId(offer.getSlaveId());
-
-      return taskBuilder.addResources(Resource.newBuilder().setName("cpus").setType(Value.Type.SCALAR).setScalar(taskCpus).build()).addResources(Resource.newBuilder().setName("mem").setType(Value.Type.SCALAR).setScalar(taskMemory).build())
-          .addResources(Resource.newBuilder().setName("ports").setType(Value.Type.RANGES).setRanges(Value.Ranges.newBuilder().addRange(Value.Range.newBuilder().setBegin(ports.getRpcPort()).setEnd(ports.getRpcPort()).build()).addRange(Value.Range
-              .newBuilder().setBegin(ports.getLocalizerPort()).setEnd(ports.getLocalizerPort()).build()).addRange(Value.Range.newBuilder().setBegin(ports.getWebAppHttpPort()).setEnd(ports.getWebAppHttpPort()).build()).addRange(Value.Range
-              .newBuilder().setBegin(ports.getShufflePort()).setEnd(ports.getShufflePort()).build()))).setExecutor(executorInfo).build();
+      TaskInfo.Builder taskBuilder = TaskInfo
+          .newBuilder()
+          .setName("task-" + taskId.getValue())
+          .setTaskId(taskId)
+          .setSlaveId(offer.getSlaveId());
+
+      return taskBuilder
+          .addAllResources(taskUtils.getScalarResource(offer, "cpus", taskCpus.getValue(), taskUtils.getExecutorCpus()))
+          .addAllResources(taskUtils.getScalarResource(offer, "mem", taskMemory.getValue(), taskUtils.getExecutorMemory()))
+          .addResources(Resource.newBuilder()
+              .setName("ports")
+              .setType(Value.Type.RANGES)
+              .setRanges(Value.Ranges.newBuilder()
+                  .addRange(Range.newBuilder()
+                      .setBegin(ports.getRpcPort())
+                      .setEnd(ports.getRpcPort())
+                      .build())
+                  .addRange(Range.newBuilder()
+                      .setBegin(ports.getLocalizerPort())
+                      .setEnd(ports.getLocalizerPort())
+                      .build())
+                  .addRange(Range.newBuilder()
+                      .setBegin(ports.getWebAppHttpPort())
+                      .setEnd(ports.getWebAppHttpPort())
+                      .build())
+                  .addRange(Range.newBuilder()
+                      .setBegin(ports.getShufflePort())
+                      .setEnd(ports.getShufflePort())
+                      .build())))
+          .setExecutor(executorInfo)
+          .build();
```

Comment:
Same as above. If this is just a formatting change, I suggest we leave it to @klucar to avoid conflicts while merging the PRs.